src/controller/java/src/chip/onboardingpayload/VendorId.kt

```diff
@@ -0,0 +1,726 @@
+/*
+ *
+ *    Copyright (c) 2023 Project CHIP Authors
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+package chip.onboardingpayload
+
+/**
+ * code="0x0000" is defined as "Matter Standard" in the official CSA alliance manufacturer ID database
+ * and is treated as invalid in Matter SDK.
+ */
+enum class VendorId(val value: Int) {
```

Comment:
This should simply not be in the SDK. We do not use any of those constants directly.

Reply:
At the same time this is moving some existing class ... maybe the removal/update should be a separate PR. I will add a checkmark, @yufengwangca let me know what you want to do - same PR or separate for not having VendorId at all.

Reply:
This emun class is used in OnboardingPayload.kt and Android CHIPTool, more will be used in QRCodeOnboardingPayload.

```
./java/src/chip/onboardingpayload/OnboardingPayload.kt:472:    if (!isVendorIdValidOperationally(vendorId) && vendorId != VendorId.UNSPECIFIED.value) {
./java/src/chip/onboardingpayload/OnboardingPayload.kt:476:    if (productId == 0 && vendorId != VendorId.UNSPECIFIED.value) {
./java/src/chip/onboardingpayload/OnboardingPayload.kt:484:    return vendorId != VendorId.UNSPECIFIED.value && vendorId <= VendorId.TESTVENDOR4.value
```

The remaining change of SetupPayload is quite big, I am still unit test it and fix various bug I found, I plan to merge the part already ready for review to make the final review easier. This change is only about relocation, no logic change.

Reply:
This change is needed so that those enum definition can be share used by QRCodeOnboardingPayload later

Reply:
@yufengwangca  assuming that these comments mean `separate PR`

The point is that all of the usages currently only care about 0-value and max-value, so having different constants is generally useless. These should be just a integer (or maybe a wrapped integer for type safety) and we can remove the actual vendorid constants. Constants will just move to be out of date.